l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -168,24 +206,32 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
             uint256 balanceBefore = address(this).balance;
             IMailbox(_target).transferEthToSharedBridge();
             uint256 balanceAfter = address(this).balance;
-            require(balanceAfter > balanceBefore, "ShB: 0 eth transferred");
+            if (balanceAfter <= balanceBefore) {
+                revert NoFundsTransferred();
+            }
             chainBalance[_targetChainId][ETH_TOKEN_ADDRESS] =
                 chainBalance[_targetChainId][ETH_TOKEN_ADDRESS] +
                 balanceAfter -
                 balanceBefore;
         } else {
             uint256 balanceBefore = IERC20(_token).balanceOf(address(this));
             uint256 legacyBridgeBalance = IERC20(_token).balanceOf(address(legacyBridge));
-            require(legacyBridgeBalance > 0, "ShB: 0 amount to transfer");
+            if (legacyBridgeBalance == 0) {
+                revert ZeroBalance();
+            }
             IL1ERC20Bridge(_target).transferTokenToSharedBridge(_token);
             uint256 balanceAfter = IERC20(_token).balanceOf(address(this));
-            require(balanceAfter - balanceBefore == legacyBridgeBalance, "ShB: wrong amount transferred");
+            if (balanceAfter - balanceBefore != legacyBridgeBalance) {
+                revert ValueMismatch(balanceAfter - balanceBefore, legacyBridgeBalance);
+            }
             chainBalance[_targetChainId][_token] = chainBalance[_targetChainId][_token] + legacyBridgeBalance;
         }
     }
 
     function receiveEth(uint256 _chainId) external payable {
-        require(BRIDGE_HUB.getHyperchain(_chainId) == msg.sender, "receiveEth not state transition");
+        if (BRIDGE_HUB.getHyperchain(_chainId) != msg.sender) {
+            revert RevertWithMsg(ERR_RECEIVE_ETH_BRIDGE);
```

Comment:
Why use message here?

Reply:
it seemed to be a hyper-specific error but I can change it out